src/services/verify/index.tsx

```diff
@@ -7,21 +7,22 @@ import {
 } from "@govtechsg/oa-verify";
 import { providers } from "ethers";
 import { NETWORK_NAME } from "../../config";
+import { getCurrentProvider } from "../../common/contexts/provider";
 
 export enum VerifierType {
   DEMO = "demo",
   CUSTOM = "custom",
 }
 
-const verificationOption =
-  NETWORK_NAME === "local"
+const verificationOption = (provider: providers.Provider | undefined) =>
+  provider
+    ? { provider }
+    : NETWORK_NAME === "local"
     ? { provider: new providers.JsonRpcProvider(), network: NETWORK_NAME }
     : { network: NETWORK_NAME };
```

Comment:
okay to structure this into a if else block instead of multiple ternaries?

Reply:
Yup, sure! Changed to multiple ternaries in https://github.com/TradeTrust/tradetrust-website/pull/506/commits/e967a113f90dcc287e323a13f25550b32f84cfc3